client/session.go

```diff
@@ -1695,9 +1797,10 @@ type blockMetadataReattempt struct {
 }
 
 type blockMetadataReattemptPeerMetadata struct {
-	peer  hostQueue
-	start time.Time
-	size  int64
+	peer     hostQueue
+	start    time.Time
+	size     int64
+	checksum *uint32
```

Comment:
Correct right now we don't use checksums to select the best peer, but I was thinking we might need to use that info later, e.g., when the peer checksums disagree, we might want to pick the peers whose checksums are the majority.